circle.yml

```diff
@@ -89,13 +89,49 @@ defaults:
         $GCOV --version
         codecov --required --gcov-exec "$GCOV" --gcov-root ~/build
 
+  save-ccache: &save-ccache
+    save_cache:
+      name: "Save ccache"
+      key: ccache-1-{{arch}}-{{checksum "compiler.version"}}-{{.Branch}}-{{epoch}}
+      paths:
+        - ~/.ccache
+
+  restore-ccache: &restore-ccache
+    restore_cache:
+      name: "Restore ccache"
+      keys:
+        - ccache-1-{{arch}}-{{checksum "compiler.version"}}-{{.Branch}}
+        - ccache-1-{{arch}}-{{checksum "compiler.version"}}
```

Comment:
The bad thing about this fallback key is that it will be soon horribly outdated. It might make more sense to use the current week or something as another level. On the other hand, perhaps you prefer full builds for the first build in a PR anyway.

Reply:
I believe it works in a different way. The Circle CI documentation is not clear about it, but from their and other examples it looks the cache key matching is prefix based and the newest matched cache blog is taken.

First of all, we update the ccache cache on every build (see `-{{epoch}}` suffix in the `save-cache`).

Here, we restore the newest ccache cache matching the current branch (I'm not sure what the "branch" means in foreign PRs). If nothing found, the newest ccache cache is taken considering all branches (including all PRs).

Considering all this, maybe better strategy would be to fallback to develop branch only, not to other branches.

I took it from https://github.com/hyperledger/quilt/blob/master/.circleci/config.yml, the only good example I've found with ccache config for Circle CI.